tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Ricci.cpp

```diff
@@ -0,0 +1,134 @@
+          make_deriv_spatial_christoffel_second_kind<spatial_dim>(
+              used_for_size)),
+      ricci);
+}
+
+void test_1d_spacetime_ricci(const DataVector& used_for_size) {
+  const size_t spatial_dim = 1;
+  const auto ricci = gr::ricci_tensor(
+      make_spacetime_christoffel_second_kind<spatial_dim>(0.),
+      make_deriv_spacetime_christoffel_second_kind<spatial_dim>(0.));
+  CHECK(ricci.get(0, 0) == approx(-46.));
+  CHECK(ricci.get(0, 1) == approx(28.));
+  CHECK(ricci.get(1, 1) == approx(-16.));
+
+  check_tensor_doubles_equals_tensor_datavectors(
+      gr::ricci_tensor(
+          make_spacetime_christoffel_second_kind<spatial_dim>(used_for_size),
+          make_deriv_spacetime_christoffel_second_kind<spatial_dim>(
+              used_for_size)),
+      ricci);
+}
+
+void test_2d_spacetime_ricci(const DataVector& used_for_size) {
+  const size_t spatial_dim = 2;
+  const auto ricci = gr::ricci_tensor(
+      make_spacetime_christoffel_second_kind<spatial_dim>(0.),
+      make_deriv_spacetime_christoffel_second_kind<spatial_dim>(0.));
+  CHECK(ricci.get(0, 0) == approx(-406.));
+  CHECK(ricci.get(0, 1) == approx(-32.));
+  CHECK(ricci.get(0, 2) == approx(217.));
+  CHECK(ricci.get(1, 1) == approx(-178.));
+  CHECK(ricci.get(1, 2) == approx(143.5));
+  CHECK(ricci.get(2, 2) == approx(-201.));
+
+  check_tensor_doubles_equals_tensor_datavectors(
+      gr::ricci_tensor(
+          make_spacetime_christoffel_second_kind<spatial_dim>(used_for_size),
+          make_deriv_spacetime_christoffel_second_kind<spatial_dim>(
+              used_for_size)),
+      ricci);
+}
+
+void test_3d_spacetime_ricci(const DataVector& used_for_size) {
+  const size_t spatial_dim = 3;
+  const auto ricci = gr::ricci_tensor(
+      make_spacetime_christoffel_second_kind<spatial_dim>(0.),
+      make_deriv_spacetime_christoffel_second_kind<spatial_dim>(0.));
+  CHECK(ricci.get(0, 0) == approx(-1928.));
+  CHECK(ricci.get(0, 1) == approx(-700.));
+  CHECK(ricci.get(0, 2) == approx(283.));
+  CHECK(ricci.get(0, 3) == approx(940.));
+  CHECK(ricci.get(1, 1) == approx(-1300.));
+  CHECK(ricci.get(1, 2) == approx(82.5));
+  CHECK(ricci.get(1, 3) == approx(968.));
+  CHECK(ricci.get(2, 2) == approx(-629.));
+  CHECK(ricci.get(2, 3) == approx(335.5));
+  CHECK(ricci.get(3, 3) == approx(-1176.));
+
+  check_tensor_doubles_equals_tensor_datavectors(
+      gr::ricci_tensor(
+          make_spacetime_christoffel_second_kind<spatial_dim>(used_for_size),
+          make_deriv_spacetime_christoffel_second_kind<spatial_dim>(
+              used_for_size)),
+      ricci);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.Ricci.",
+                  "[PointwiseFunctions][Unit]") {
+  const DataVector dv(2);
+  test_1d_spatial_ricci(dv);
+  test_2d_spatial_ricci(dv);
+  test_3d_spatial_ricci(dv);
+  test_1d_spacetime_ricci(dv);
+  test_2d_spacetime_ricci(dv);
+  test_3d_spacetime_ricci(dv);
```

Comment:
Similar call all these with a double

Reply:
Please clarify what you are asking for and why? These functions only take DataVectors. Test_Ricci.cpp does exactly what Test_Christoffel does, so I don't understand what more is necessary or desirable here. Are you asking me to make a copy of all these functions that takes doubles instead of data vectors?

Reply:
after templating above, you now need to call them separately with a double and a DataVector